### PR TITLE
Update ns-System.Net.Mail.xml

### PR DESCRIPTION
**Pull request by stephenegriffin**
Lift the warning from SmtpClient to the namespace to make it clearer this is now obsolete.

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: xml/ns-System.Net.Mail.xml [hunk 1]
```diff
@@ -1,12 +1,20 @@
 <Namespace Name="System.Net.Mail">
+  <Attributes>
```
Comment (rpetrusha) [inline xml/ns-System.Net.Mail.xml]: This is a reflection-only portion of the XML. Please revert this change for the new lines 2 through 6.

File: xml/ns-System.Net.Mail.xml [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
     <summary>The <see cref="N:System.Net.Mail" /> namespace contains classes used to send electronic mail to a Simple Mail Transfer Protocol (SMTP) server for delivery.</summary>
     <remarks>
       <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
  The <xref:System.Net.Mail.MailMessage> class represents the content of a mail message. The <xref:System.Net.Mail.SmtpClient> class transmits email to the SMTP host that you designate for mail delivery. You can create mail attachments using the <xref:System.Net.Mail.Attachment> class.  
-  
+
+  [!IMPORTANT] 
+  We don't recommend that you use the `SmtpClient` class for new development. For more information, see [SmtpClient shouldn't be used](https://github.com/dotnet/platform-compat/blob/master/docs/DE0005.md) on GitHub.
```
Comment (rpetrusha) [inline xml/ns-System.Net.Mail.xml]: The formatting here is incorrect to produce a note box. The lines should be preceded by `> `.  But this seems an awkward way of handling the fact that SmtpClient is obsolete in any case. Instead of the note, I'd modify the Remarks section as I've suggested, and also add "(This type is now obsolete.)" to the type summary in dotnet-api-docs/xml/System.Net.Mail/SmtpClient.xml.

File: xml/ns-System.Net.Mail.xml [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+      <AttributeName>System.Obsolete("SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead")</AttributeName>
+    </Attribute>
+  </Attributes>
   <Docs>
     <summary>The <see cref="N:System.Net.Mail" /> namespace contains classes used to send electronic mail to a Simple Mail Transfer Protocol (SMTP) server for delivery.</summary>
     <remarks>
       <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
  The <xref:System.Net.Mail.MailMessage> class represents the content of a mail message. The <xref:System.Net.Mail.SmtpClient> class transmits email to the SMTP host that you designate for mail delivery. You can create mail attachments using the <xref:System.Net.Mail.Attachment> class.
```
Comment (rpetrusha) [inline xml/ns-System.Net.Mail.xml]:
```suggestion
 The <xref:System.Net.Mail.MailMessage> class represents the content of a mail message. 
 
 You can create mail attachments using the <xref:System.Net.Mail.Attachment> class.  
```

**Comment by rpetrusha** [comment]
Since you didn't respond, @stephenegriffin, I've incorporated my suggestions into your PR, and also opened #2483 to change the description of SmtpClient. I'll merge your PR now.

**Comment by stephenegriffin** [comment]
Sorry - missed your reply - thanks for following through on this.

**Comment by stephenegriffin** [comment]
Actually - this is all wrong - the point was to indicate that the entire System.Net.Mail namespace is deprecated. You've removed that and added a redundant note to SmtpClient.
